org.scala-ide.sdt.core/src/org/scalaide/extensions/autoedits/SurroundBlock.scala

```diff
@@ -92,10 +102,8 @@ trait SurroundBlock extends AutoEdit {
 
             if (prevLine.start == firstLine.start)
               None
-            else if (t.tokenType == Tokens.VARID)
```

Comment:
This part I didn't understand what it was about. After removing it, no test failed, and it's what helped fix my bug about including blank lines in surround block.